core/worker.cc

```diff
@@ -300,7 +300,7 @@ void *Worker::Run(void *_arg) {
 
   current_tsc_ = rdtsc();
 
-  pframe_pool_ = bess::get_pframe_pool();
+  pframe_pool_ = bess::get_pframe_pool_socket(socket_);
```

Comment:
Should this be a separate thing? It's good to be NUMA aware here but isn't that independent of the thread-local cleanup...?